packages/expo-av/ios/EXAV/Video/EXVideoView.m

```diff
@@ -472,6 +491,11 @@ - (void)setSource:(NSDictionary *)source
 {
   if (![source isEqualToDictionary:_lastSetSource]) {
     EX_WEAKIFY(self);
+    // ? Why dispatch to _exAV.methodQueue rather than remain on the main thread?
+    // ? Can lead to race conditions with Imperative API being sent on the main thread.
+    // ? I've made Imperative API dispatch to _exAV.methodQueue since I do not know
+    // ? the reason for it, but ultimately I believe Prop setters should run on the main thread
+    // ? rather than move Imperative API methods to _exAV.methodQueue.
```

Comment:
I've got absolutely no idea why it was designed that way 🤔 
I guess somebody tried to limit the amount of work main thread is doing in this module, but I don't have a clue why 🤷

Reply:
Thank you for the response,

I made the `EXAV` react native module run on the main thread so that we can hopefully avoid all of these memory-related crashes and stop dealing with them on a case-by-case basis.

If it turns out that there is some heavy processing needed, we can always dispatch that to a separate queue and then dispatch the results back to the main thread.